### DISCUSSION TITLE
Ruby 3.2.1 build failed (Ubuntu 20.04)

### DISCUSSION THREAD

**Discussion by andreyruby** [rbenv/ruby-build#2150] [Build failures]
Error happens only with 3.2.1 

```
rbenv install 3.2.1

=> ./tool/file2lastrev.rb:6:in `require': cannot load such file -- optparse (LoadError)
```

```
BUILD FAILED (Ubuntu 20.04 using ruby-build 20230208.1-2-g4d2e1cc)

Inspect or clean up the working tree at /tmp/ruby-build.20230211001826.752350.TsWEQE
Results logged to /tmp/ruby-build.20230211001826.752350.log

Last 10 log lines:
linking shared-object json/ext/generator.so
make[2]: Leaving directory '/tmp/ruby-build.20230211001826.752350.TsWEQE/ruby-3.2.1/ext/json/generator'
make[2]: Entering directory '/tmp/ruby-build.20230211001826.752350.TsWEQE/ruby-3.2.1'
./tool/file2lastrev.rb:6:in `require': cannot load such file -- optparse (LoadError)
	from ./tool/file2lastrev.rb:6:in `<main>'
make[2]: *** [uncommon.mk:1224: .revision.time] Error 1
make[2]: Leaving directory '/tmp/ruby-build.20230211001826.752350.TsWEQE/ruby-3.2.1'
make[1]: *** [exts.mk:93: ruby] Error 2
make[1]: Leaving directory '/tmp/ruby-build.20230211001826.752350.TsWEQE/ruby-3.2.1'
make: *** [uncommon.mk:331: build-ext] Error 2
```

**Comment by hsbt** [comment]
This is same as https://github.com/rbenv/ruby-build/discussions/1851. But I don't have any idea to resolve this yet. It may cause by `global` ruby configuration.

**Reply by andreyruby** [reply]
@hsbt 

I've  managed to install `3.2.1` after switching rbenv global to `3.1.2`. It was not install when I had rbenv global `3.2.0`

```
10:53 ~/ruby $ rbenv global 3.2.0
10:53 ~/ruby $ rbenv install 3.2.1

BUILD FAILED (Ubuntu 20.04 using ruby-build 20230208.1-2-g4d2e1cc)
./tool/file2lastrev.rb:6:in `require': cannot load such file -- optparse (LoadError)

========================================

11:04 ~/ruby $ rbenv global 3.1.2
11:05 ~/ruby $ rbenv install 3.2.1

Installed ruby-3.2.1
```

**Reply by hsbt** [reply]
Thanks for your input. It's strange result.

**Comment by zweifuchs** [comment]
I had the same problem updating from 3.3.0 to 3.3.2.
> ./tool/file2lastrev.rb:6:in `require': cannot load such file -- optparse (LoadError)

i then proceeded to install 3.1.2. That worked. Then i set 3.1.2 to global and installed 3.3.2

I did pull from the rbenv master before and also i pulled the ruby-build master branch. I still had that optparse error.

So it seems like with 3.1.2 we can install other ruby versions but not with 3.3.0. This is just for people looking for a solution to this error. My system is fedora40 and i use rbenv exclusively.

**Comment by mensfeld** [comment]
I had the same issue and was only able to fix this by installing via mise instead of rbenv. Had to nuke rbenv, resolve to system ruby on ubuntu and go via mise.
